app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# This controller deals with the cart API requests.
+class CartsController < ApplicationController
+  before_action :check_cart_params, only: :create
+
+  def create
+    cart = Cart.new(cart_params[:lineItems])
+
+    if cart.valid?
+      format_response(cart)
+    else
+      render json: { errors: cart.errors.full_messages }, status: :unprocessable_entity
+    end
+  rescue ActionController::UnpermittedParameters => e
+    render json: { error: e }, status: :unprocessable_entity
+  end
+
+  private
+
+  def cart_params
+    params.require(:cart).permit(:reference, lineItems: %i[name price collection])
+  end
+
+  def check_cart_params
```

Comment:
@victorfeijo Ideally, the model deals with business logic only, not with formatting view stuff. 

Doing that at a controller level makes more sense since params are injected into the controller. I don't see any reason to inject parameters into the model, so the model formatting them. With that said, if I had more time, I would use a `serializer` class to deal with formatting.

Not 100% sure if I understood your question. Please let me know if I didn't answer your question. ☺️

Reply:
Awesome, that makes total sense @euthayna - probably not handling on the model but could move from the controller to a service/serializer that would handle this formatting/parsing logic 👍🏽